### DISCUSSION TITLE
unpairing and re-pairing dualsense over bluetooth breaks device hiding

### DISCUSSION THREAD

**Discussion by snack0verflow** [nefarius/HidHide#63] [Q&A]
I am having this problem where if I pair the controller with a different device and pair it back to windows hidhide doesn't work, I can see the device shown in windows controller settings

**Comment by nefarius** [comment]
This is per design of Windows, if you unpair and repair a device, the Instance ID (that HidHide uses to uniquely identifies the device) changes and the match configured within HH no longer works. There is no way around this, I'm afraid.

Cheers

**Reply by Kanuan** [reply]
If the instance ID change, shouldn't HidHide be able to detect it's not the same device and then show a new entry?

**Reply by nefarius** [reply]
It should but since OP did neither provide screenshots nor more valuable information, all that is left is guesswork which I do not like to do as it's a waste of time 😉

**Reply by Kanuan** [reply]
HidHide Client won't show a new entry. The same entry will appear as if it were "locked", but that entry won't be valid for the device anymore. To fix it one need to uncheck and re-check it, which seems to not be the expected behavior

![image](https://user-images.githubusercontent.com/24910442/152377835-0e47359e-27d8-4ca6-849b-3bcff2ca1508.png)

**Reply by snack0verflow** [reply]
@nefarius , @Kanuan is correct. hidhide still shows that the device is hidden when it clearly is not there is no way to hide the device after that. I have tried restarting windows, unhiding and hiding again, running hidhide as administrator. 
Only works if I reinstall hidhide and restart, but sometimes even that fails.
This has given me much trouble.

**Reply by Kanuan** [reply]
On my end un-checking then re-checking the device as hidden fixes the issue. What version of HidHide are you using?

**Reply by snack0verflow** [reply]
1.1.50

**Reply by nefarius** [reply]
I assume te ID changed but somehow the UI associates the same name to the previous state? The CLI output would shed light on what's actually stored in the drivers setting and if this is a UI issue.

**Reply by snack0verflow** [reply]
What command's output would be useful for you from the cli when this happens the next time?

**Reply by nefarius** [reply]
```PowerShell
cd "C:\Program Files\Nefarius Software Solutions e.U\HidHideCLI"
PS C:\Program Files\Nefarius Software Solutions e.U\HidHideCLI> .\HidHideCLI.exe --dev-list
--dev-hide "HID\VID_054C&PID_0CE6&MI_03\a&c9e84b2&0&0000"
```

**Reply by Kanuan** [reply]
The CLI shows the old ID instance. If you make it list the currently present devices the same controller will have a different ID

With the issue occurring, this is how my DS4v2 as it appears when the "list connected devices" command is used (--dev-gaming)

```C
HID\{00001124-0000-1000-8000-00805f9b34fb}_VID&0002054c_PID&09cc\9&1a03a06a&18&0000
```

ANd this is how it appears when the "list HidHide's blacklist" command is used (--dev-list)
```C
HID\{00001124-0000-1000-8000-00805f9b34fb}_VID&0002054c_PID&09cc\9&1a03a06a&17&0000
```

**Reply by nefarius** [reply]
So the driver is fine, it is supposed to hold the old value, the UI is mixing it up for whatever reason, hm... 🤔
